src/components/ToastNotificationList.tsx

```diff
@@ -0,0 +1,239 @@
+    prevNotificationsSize.current = notifications.length;
+    if (!notifications.length || notificationsRemoved) {
+      return;
+    }
+
+    const container = containerRef.current;
+    const lastNotification = notifications[notifications.length - 1];
+    const notificationEl = document.getElementById(lastNotification.id);
+    if (container && notificationEl) {
+      const currentScrollY = container.scrollTop;
+      const offsetHeight =
+        notificationEl.getBoundingClientRect().height +
+        parseFloat(window.getComputedStyle(notificationEl).marginTop) +
+        parseFloat(window.getComputedStyle(notificationEl).marginBottom);
+      // only adjust the scroll height if the scroll is at the top
+      if (currentScrollY === 0) {
+        container.scrollTop = currentScrollY + offsetHeight;
+      }
+    }
+  };
+
+  const handleFilterSelect = (filter: FilterTypes) => {
+    setFilters((prevFilters) => {
+      const newFilters = new Set(prevFilters);
+      if (!newFilters.has(filter)) {
+        newFilters.add(filter);
+      } else {
+        newFilters.delete(filter);
+      }
+      return newFilters;
+    });
+  };
+
+  const handleGroupedDismiss = () => {
+    if (hasFilters) {
+      const notificationsToClear = notifications.filter((notification) =>
+        filters.has(notification.type),
+      );
+      onDismiss(notificationsToClear);
+      setFilters(new Set());
+      return;
+    }
+
+    onDismiss();
+  };
+
+  const getSeverityFilters = () => {
+    const filterButtons = severityOrder.map((severity) => {
+      if (groupedCount[severity]) {
+        return (
+          <button
+            aria-label={`Filter ${severity} notifications`}
+            aria-pressed={filters.has(severity)}
+            key={severity}
+            className="u-no-margin u-no-border filter-button"
+            onClick={() => handleFilterSelect(severity)}
+          >
+            <Icon name={iconLookup[severity]} />
+            <span>{groupedCount[severity]}</span>
+          </button>
+        );
+      }
+    });
+
+    return (
+      <div className="filters">
+        {filterButtons}
+        {hasFilters && (
+          <button
+            className="u-no-margin--bottom u-no-border"
+            onClick={() => setFilters(new Set())}
+          >
+            Clear filters
+          </button>
+        )}
+      </div>
+    );
+  };
+
+  const getDismissText = () => {
+    if (hasFilters) {
+      const validFilters = Object.keys(groupedCount) as FilterTypes[];
+      let totalCount = 0;
+      for (const filter of validFilters) {
+        if (filters.has(filter)) {
+          totalCount += groupedCount[filter] || 0;
+        }
+      }
+
+      const dismissText = (
+        <span className="dismiss-text">Dismiss {totalCount}</span>
+      );
+      return dismissText;
+    }
+
+    return <span>Dismiss all</span>;
+  };
+
+  const handleDismissNotification = (notification: ToastNotificationType) => {
```

Comment:
@lorumic or @aaryanporwal , this code here is used to animate dismissing a single notification from the notification list. The notification will fade out and have its height reduced to 0 before being unmounted from the DOM. In addition, if the user has the `prefers-reduced-motion` setting enabled on their OS, then no animation will be shown. You can emulate `prefers-reduced-motion` in chrome using this [guide](https://developer.chrome.com/docs/devtools/rendering/emulate-css#emulate_css_media_feature_prefers-reduced-motion).